android/src/main/java/com/rnfingerprint/FingerprintDialog.java

```diff
@@ -0,0 +1,142 @@
+package com.rnfingerprint;
+
+import android.app.DialogFragment;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import java.lang.String;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.content.DialogInterface;
+import android.view.KeyEvent;
+import android.graphics.Color;
+import com.facebook.react.bridge.ReadableMap;
+import android.hardware.fingerprint.FingerprintManager;
+import android.os.Handler.Callback;
+
+public class FingerprintDialog extends DialogFragment
+        implements FingerprintHandler.Callback {
+
+    private Button mCancelButton;
+    private View mFingerprintContent;
+    private TextView mFingerprintDescription;
+    private ImageView mFingerprintImage;
+
+    private FingerprintManager.CryptoObject mCryptoObject;
+    private DialogResultListener dialogCallback;
+    private FingerprintHandler mFingerprintHandler;
+
+    private String authReason;
+    private ReadableMap authConfig;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Do not create a new Fragment when the Activity is re-created such as orientation changes.
+        setRetainInstance(true);
```

Comment:
This is considered a bad practice. You should only retain a fragment for a dialog that returns no view, so `onCreateView` should return null. Anyway you already use a DialogFragment (they don't leak on rotation), and RN in manifest also says that do not destroy the activity so you do not need this line.